go/common/node/address.go

```diff
@@ -137,6 +138,45 @@ func toProtoAddress(addr Address) *pbCommon.Address {
 	return pbAddr
 }
 
+// ConsensusAddress contains the ID and address for connecting to a consensus
+// member.
+// NOTE: This could be the P2P ID and IP of a sentry node.
+type ConsensusAddress struct {
+	ID      signature.PublicKey `json:"id"`
+	Address Address             `json:"address"`
```

Comment:
General question...

Unless I'm mistake, the tendermint protocol sentry nodes as used for cosmos are meant to be a dynamic set of machines. These sentry node machines could in theory continuously disappear and reappear with different IP addresses and identities. The gossip nature of the tendermint protocol would allow this to work. Is there a reason that we need to specify more than the `ID` when registering a node for consensus?

cc @Yawning @kostko

Reply:
Yeah this would probably work as well, but would rely exclusively on the seed nodes to establish peers via PEX.

Also tendermint AFAIK requires you to specify validator addresses in the genesis block?

Reply:
Also "sentry nodes" (aka proxies) for other things (gRPC and possibly libp2p) will definitely require addresses and we should keep that in mind while designing any proxy stuff.

Reply:
> Also tendermint AFAIK requires you to specify validator addresses in the genesis block?

Strictly speaking, no: https://godoc.org/github.com/tendermint/tendermint/types#GenesisValidator

Our tendermint seed node implementation currently uses the genesis document to populate the initial address book.  This can be omitted, assuming "enough" nodes use the seed node.  Time to a functional network after redeploy will be longer.

From a practical perspective, we use more than tendermint, and I don't see us implementing a gossip protocol anytime soon.  So I'm not sure what this saves if anything.

Reply:
As long as it's possible for the tendermint sentry node to disappear entirely, I would say whatever gets is alright. 

> Also "sentry nodes" (aka proxies) for other things (gRPC and possibly libp2p) will definitely require addresses and we should keep that in mind while designing any proxy stuff.

I would say registering an address for libp2p/grpc proxy would be fine. Would it be possible to register a DNS name and cert for those proxies so that IPs can float and be scaled out?

Reply:
> Would it be possible to register a DNS name and cert for those proxies so that IPs can float and be scaled out?

Yes.